list/list.go

```diff
@@ -255,12 +256,40 @@ func (m *Model) SetShowTitle(v bool) {
 	m.updatePagination()
 }
 
+// SetFilterText explicitly sets the filter text without relying on user input.
+// It also sets the filterState to a sane default of FilterApplied, but this
+// can be changed with SetFilterState
+func (m *Model) SetFilterText(filter string) {
+	m.filterState = Filtering
+	m.FilterInput.SetValue(filter)
+	cmd := filterItems(*m)
```

Comment:
Should this maybe be returned by the functions? If a command is sent by a function, surely it might be prudent to send that along?

Reply:
Basically `SetFilterText()` should return a `tea.Cmd`?

Reply:
No, we call it and handle it. The model might not have been initialized yet. We can't assume the program is already running.

Reply:
ah ok that makes sense.